src/main/protobuf/bazel_output_service.proto

```diff
@@ -0,0 +1,324 @@
+  // Intentionally empty for future extensibility.
+}
+
+message StartBuildRequest {
+  // The version of the protocol Bazel currently uses. The service MUST return
+  // an error if it doesn't recognize the version.
+  //
+  // A future Bazel version may introduce incompatible changes and increment
+  // this version number. The incompatible change will be first made in the
+  // development tree for the next major Bazel release, and the new version thus
+  // introduced should be considered unstable until that major Bazel release
+  // occurs. At that time, the new version becomes stable, and the old one is
+  // immediately retired.
+  //
+  // In particular, version 1 must not be considered stable until Bazel 8.0.0 is
+  // released.
+  //
+  // Current version: 1 (experimental).
+  int32 version = 1;
+
+  // A client-chosen value for the output service to uniquely identify the
+  // workspace the build is being started. This value must be set to ensure that
+  // the remote output service is capable of managing builds for distinct
+  // workspaces concurrently.
+  //
+  // Bazel sets this value to the MD5 sum of the absolute path of the output
+  // base.
+  //
+  // Starting a build finalizes any previous build with the same output_base_id
+  // that has not been finalized yet as if a FinalizeBuildRequest had been sent
+  // with build_successful set to false.
+  string output_base_id = 2;
+
+  // A client-chosen value that uniquely identifies this build. This value must
+  // be provided to most other methods to ensure that operations are targeted
+  // against the right output path.
+  //
+  // Bazel sets this value to --invocation_id.
+  string build_id = 3;
+
+  // Additional arguments to pass depending on how Bazel communicate with the
+  // Content Addressable Storage.
+  google.protobuf.Any args = 4;
+
+  // The absolute path at which the remote output service exposes its output
+  // paths, as seen from the perspective of the client.
+  //
+  // This value needs to be provided by the client, because file system
+  // namespace virtualization may cause this directory to appear at a location
+  // that differs from the one used by the service.
+  //
+  // The purpose of this field is to ensure that the remote output service is
+  // capable of expanding symbolic links containing absolute paths.
+  //
+  // If this is not set, or an empty string, the service must determine where to
+  // expose its output path and return an absolute path in
+  // StartBuildResponse.output_path_suffix.
+  string output_path_prefix = 5;
+
+  // A map of paths on the system that will become symbolic links pointing to
+  // locations inside the output path. Similar to output_path_prefix, this
+  // option is used to ensure the remote output service is capable of expanding
+  // symbolic links.
+  //
+  // Map keys are absolute paths, while map values are paths that are
+  // relative to the output path.
+  map<string, string> output_path_aliases = 6;
```

Comment:
I'm still a bit on the fence about the benefit/complexity tradeoff for this one; was there some prior discussion where we determined that this is absolutely required for v1?

If we do decide to keep it, I think we also need to explicitly state the implications of this field for the path canonicalization operation entailed by a BatchStat (I made a comment there that essentially assumes this field doesn't exist, because I only noticed it later.)